docker/Makefile

```diff
@@ -112,6 +121,12 @@ stop_client :
 # performance requirements are relatively low.
 # -----------------------------------------------------------------
 repository :
+	# if an enclave signing key is available on the host, copy that under build/keys in the repo
+	# Note: the docker build (see PDO_ENCLAVE_CODE_SIGN_PEM in environment.sh) expects the key there
+	[ ! -e ${PDO_ENCLAVE_CODE_SIGN_PEM} ] ||\
```

Comment:
Test also whether PDO_ENCLAVE_CODE_SIGN_PEM is actually defined at all!   Note in previous version you did not have to define _any_ env variable to build docker ....

Reply:
Two comments here:
1. I have added some checks to fail more gracefully and see what's going on
2. however, according to our documentation `build/common-config.sh` should be sourced first when developing in PDO. Hence,
  a. PDO_ENCLAVE_CODE_SIGN_PEM is always defined (either by the user, or by the script)
  b. it would be more effective to check whether the script was sourced -- I'm open to suggestions

Reply:
> 2\. however, according to our documentation, `build/common-config.sh` should be sourced first when developing in PDO. Hence,

This is not really required for docker (and from the docker docu also not obvious)

Reply:
> why do we insist that it has to be in ${PDO_SGX_KEY_ROOT} (outside of docker) and don't just copy it to xfer directly?

(Good question!) From previous discussions with Mic, we noticed two things:
1. the enclave code signing key right now is only needed at build time, and if one is available we should pass it into the build
2. the build happens while the docker image is created, and we cannot mount the xfer folder at that time.

For these reasons, we decided to make ${PDO_SGX_KEY_ROOT} (set to its default value, and thus within the repo!) as the destination folder for the key. In this way, the key copied/cloned together with the repo in the docker container. Then, at build time within the container, the common-config is sourced (setting the default values), and the build will find the key in the right place.

Since the built image is meant to be distributed, in the future we will have to remove the key material. We can accomplish that by removing the cloned repo (which includes the key).

Reply:
Ah, i see.  But might be worth mentioning above as a comment in that code fragment as not so obvious?